File: src/main/java/grojdg/harderstats/PlayerStats.java
package grojdg.harderstats;

public class PlayerStats {
    private int damageTaken = 0;

    private int mobsKilled = 0;

    private int foodEaten = 0;

    private int experienceGained = 0;

    public StatTimer waterTimer = new StatTimer();

    public StatTimer netherTimer = new StatTimer();

    public void updateDamageTaken(int damage) {
        damageTaken += damage;
    }

    public void updateMobsKilled(int count) {
        mobsKilled += count;
    }

    public void updateFoodEaten(int count) {
        foodEaten += count;
    }

    public void updateExperiencedGained(int experience) {
        experienceGained += experience;
    }

    public int getDamageTaken() {
        return damageTaken;
    }

    public int getMobsKilled() {
        return mobsKilled;
    }

    public int getFoodEaten() {
        return foodEaten;
    }

    public int getExperienceGained() {
        return experienceGained;
    }

    public void resetDamageTaken() {
        damageTaken = 0;
    }

    public void resetMobsKilled() {
        mobsKilled = 0;
    }

    public void resetFoodEaten() {
        foodEaten = 0;
    }

    public void resetExperienceGained() {
        experienceGained = 0;
    }
}

File: src/main/kotlin/grojdg/harderstats/PlayerStats.kt
package grojdg.harderstats

class PlayerStats {
    var damageTaken = 0
        private set
    var mobsKilled = 0
        private set
    var foodEaten = 0
        private set
    var experienceGained = 0
        private set
    var waterTimer: StatTimer = StatTimer()
    var netherTimer: StatTimer = StatTimer()
    fun updateDamageTaken(damage: Int) {
        damageTaken += damage
    }

    fun updateMobsKilled(count: Int) {
        mobsKilled += count
    }

    fun updateFoodEaten(count: Int) {
        foodEaten += count
    }

    fun updateExperiencedGained(experience: Int) {
        experienceGained += experience
    }

    fun resetDamageTaken() {
        damageTaken = 0
    }

    fun resetMobsKilled() {
        mobsKilled = 0
    }

    fun resetFoodEaten() {
        foodEaten = 0
    }

    fun resetExperienceGained() {
        experienceGained = 0
    }
}
